### PR TITLE
Determine start pipeline channel

### PR DESCRIPTION
**Pull request by fgimenez**
Towards https://github.com/giantswarm/giantswarm/issues/1902

Determines initial the name off the initial channel in the pipeline from the chart version.

### CODE REVIEW

**Comment by fgimenez** [comment]
Sorry for not being more explicit in the description, I've added comments to the `pipeline` package and `StartChannel` method hoping to make things more clear.

The pipeline metaphor is implied in the delivery model defined in https://github.com/giantswarm/giantswarm/blob/master/roadmap/MANAGED-SERVICES.md, the charts are going through different stages and we do different things at each of them, initial build, automated testing, deploy to a staging environment, deploy to production, with manual/automated approvals in between. This is what I see like a pipeline, it is a concept commonly used in CI/CD systems, maybe we could use some other abstraction here, or mention it in the spec, let me know WDYT.

This PR implements the initial part of this sequence of steps, it pushes the chart to the first channel (`X-Y-beta` in the spec). It doesn't do anything else by itself, there can be other systems listening to CNR that can react to this change, AIUI this will trigger other actions according to the loosely coupled architecture defined by the spec, pls correct me if I'm wrong :)

**Comment by fgimenez** [comment]
Thanks!

**Comment by rossf7** [comment]
Thanks for the GoDoc comments and the extra explanation here. Adding the pipelines concept makes total sense and is aligned with the spec.

**Comment by fgimenez** [comment]
Indeed, AIUI it's already described here https://github.com/giantswarm/giantswarm/blob/master/roadmap/MANAGED-SERVICES.md#release-promotion, this is the same as a delivery pipeline (not deployment pipeline because of the potential manual steps).

File: pipeline/pipeline.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	elements := strings.Split(c.Version, ".")
+	if len(elements) < 2 {
+		return "", microerror.Mask(incorrectChartVersionError)
+	}
+	// remove initial v
+	elements[0] = strings.TrimLeft(elements[0], "v")
```
[the remark this answers is not in the document]
Comment (fgimenez) [reply pipeline/pipeline.go]: Take a look at the tests, they document the intended behavior https://github.com/giantswarm/architect/blob/master/pipeline/pipeline_test.go#L32-L64

**Pull request merged**